### PR TITLE
Bitbucket server

### PR DESCRIPTION
**Pull request by StarfallProjects**
Resolves https://trello.com/c/uC0RrxXe/190-bitbucket-server

### CODE REVIEW

File: pages/integrations/bitbucket.md.erb [hunk 1]
```diff
@@ -4,7 +4,7 @@ Buildkite integrates with [Bitbucket](https://bitbucket.org/) to provide automat
 
 <%= image "build-status.png", width: 1160/2, height: 512/2, alt: "Screenshot of the Bitbucket builds popup" %>
 
-This guide shows you how to setup your BitBucket builds with Buildkite.
+This guide shows you how to set up your BitBucket builds with Buildkite.
```
Comment (plaindocs) [inline pages/integrations/bitbucket.md.erb:7]: Bitbucket or Bitbucket?

File: pages/integrations/github_enterprise.md.erb [hunk 2]
```diff
@@ -4,25 +4,25 @@ Buildkite can connect to your GitHub Enterprise Server and use the Commit Status
 
 {:toc}
 
-## Step 1. Register Buildkite as an OAuth App
+## Step 1: Register Buildkite as an OAuth App
```
Comment (plaindocs) [inline pages/integrations/github_enterprise.md.erb:7]: Nice bunch of fixes to GitHub Enterprise. :+1:

**Review by plaindocs — COMMENTED** [review]
This is great progress! I've left a few suggestions and questions.

And I totally forgot  to mention this manual list as well. https://github.com/buildkite/docs/blob/a1ca405664baaca1eacbe04b2e45a70bba17a2c2/app/views/pages/index.html.erb#L28-L34

File: pages/integrations/bitbucket.md.erb (hunk 1 above)
Comment (StarfallProjects) [reply pages/integrations/bitbucket.md.erb:7]: Fixed

File: pages/integrations/bitbucket_server.md.erb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+2. Select *Bitbucket Server*.
+3. In *URLs*, enter the address of your Bitbucket Server, including a port if needed. For example, if you have it installed locally and running on `localhost:8000`, enter `localhost:8000`. You also have the option to add *Allowed IP Addresses* in *Network Settings*.
+4. Select *Save Settings*.
+5. Set up a pipeline as normal. Refer to [Pipelines](/docs/pipelines) for more information.
+
+## Step 2: confirm your setup
+
+If your configuration worked, Buildkite automatically recognizes your repository URL as a Bitbucket Server repository. To check this, go to *Pipelines* > *`your-pipeline`* > *Pipeline Settings*. You should see *Bitbucket Server* in the *Pipeline Settings* menu.
+
+## Step 3: work through the in-app guide to set up your webhook
```
[the remark this answers is not in the document]
Comment (StarfallProjects) [reply pages/integrations/bitbucket_server.md.erb:23]: Added a few more "Bitbucket Servers" and a sentence to the step intro to clarify

File: pages/integrations/bitbucket_server.md.erb [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+Buildkite integrates with Bitbucket Server to provide automated builds based on your source control. Your code pushes to BitBucket Server can automatically trigger Buildkite builds using a webhook.
+
+<div class="Docs__note">
+This guide shows you how to set up your BitBucket Server builds with Buildkite. It was written using Bitbucket server version 7.11.1
+</div>
+
+{:toc}
+
+## Step 1: connect a repository provider and set up a pipeline
```
[the remark this answers is not in the document]
Comment (StarfallProjects) [reply pages/integrations/bitbucket_server.md.erb:11]: Fixed

File: pages/integrations/bitbucket_server.md.erb [hunk 5]
```diff
@@ -0,0 +1,34 @@
+# Bitbucket Server
+
+Buildkite integrates with Bitbucket Server to provide automated builds based on your source control. Your code pushes to BitBucket Server can automatically trigger Buildkite builds using a webhook.
```
[the remark this answers is not in the document]
Comment (StarfallProjects) [reply pages/integrations/bitbucket_server.md.erb:3]: Fixed

**Comment by StarfallProjects** [comment]
@plaindocs all issues should now be fixed

File: pages/integrations/bitbucket_server.md.erb [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+<div class="Docs__note">
+This guide shows you how to set up your BitBucket Server builds with Buildkite. It was written using Bitbucket server version 7.11.1
+</div>
+
+{:toc}
+
+## Step 1: connect Bitbucket Server and set up a pipeline
+
+1. In your *Settings*, navigate to *Repository Providers*.
```
Comment (plaindocs) [inline pages/integrations/bitbucket_server.md.erb:13]: Style guide says we should not punctuate these.

Comment (plaindocs) [reply pages/integrations/bitbucket_server.md.erb:13]: We should distinguish between person and org settings

image

**Review by plaindocs — CHANGES_REQUESTED** [review]
Couple more tiny changes.

File: pages/integrations/bitbucket_server.md.erb [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+</div>
+
+{:toc}
+
+## Step 1: connect Bitbucket Server and set up a pipeline
+
+1. Select *Settings* to open the **Organization Settings** page.
+2. Navigate to *Repository Providers*.
+3. Select *Bitbucket Server*.
+4. In *URLs*, enter the address of your Bitbucket Server, including a port if needed. For example, `localhost:8000`. You also have the option to add *Allowed IP Addresses* in *Network Settings*.
```
Comment (plaindocs) [inline pages/integrations/bitbucket_server.md.erb:16]:
```suggestion
4. In *URLs*, enter the address of your Bitbucket Server, including a port if needed. For example, `localhost:8000`. You can also restrict which network addresses are allowed to trigger builds via webhooks in  *Allowed IP Addresses* in *Network Settings*.
```

**Review by plaindocs — APPROVED** [review]
Looks good to me. This fixes https://github.com/buildkite/docs/issues/423 from Feb 2019!!!

**Pull request merged**